integration/src/test/scala/ammonite/integration/BasicTests.scala

```diff
@@ -41,6 +41,46 @@ object BasicTests extends TestSuite{
         assert(evaled.out.trim == "Script Worked!!" && evaled.err.string.isEmpty)
       }
     }
+    'scalacNotLoadedByCachedScripts{
+      val evaled1 = %%bash(
+        executable,
+        replStandaloneResources/'basic/"Print.sc",
+        "-h",
+        "target/tempAmmoniteHome",
+        JAVA_OPTS = "-verbose:class"
+        )
+      val evaled2 = %%bash(
+        executable,
+        replStandaloneResources/'basic/"Print.sc",
+        "-h",
+        "target/tempAmmoniteHome",
+        JAVA_OPTS = "-verbose:class"
+      )
+      rm(cwd/'target/'tempAmmoniteHome)
+      val count1 = substrCount(evaled1.out.trim, "scala.tools.nsc")
+      val count2 = substrCount(evaled2.out.trim, "scala.tools.nsc")
+      assert(count1 > count2)
```

Comment:
Can we add some concrete limit to the number of classes? e.g. `count1 > 100 && count2 < 10` or something similar